app/proc/telegram.go

```diff
@@ -20,12 +24,23 @@ const maxTelegramFileSize = 50_000_000
 
 // TelegramClient client
 type TelegramClient struct {
-	Bot     *tb.Bot
-	Timeout time.Duration
+	Bot            *tb.Bot
+	Timeout        time.Duration
+	uploaderConfig *TelegramUploaderConfig
+}
+
+// TelegramUploaderConfig struct used to configure
+// experimental uploader to Telegram for large audio files
+type TelegramUploaderConfig struct {
+	Enabled      bool   `long:"uploader_enabled" env:"UPLOADER_ENABLED" description:"Enables experimental Telegram large files uploader"`
+	PathToScript string `long:"path" env:"PATH_TO_SCRIPT" description:"Path to Python uploader script"`
+	APIID        string `long:"api_id" env:"API_ID" description:"Telegram APP API ID in format like 0000000"`
+	APIHash      string `long:"api_hash" env:"API_HASH" description:"Telegram APP API Hash in format like 0123456789acbdefghijklmnopqrstuw"`
+	Session      string `long:"session" env:"SESSION" description:"Path to Telegram client session file (created by uploader/auth.py script)"`
```

Comment:
not sure why auth.py a separate thing and why the caller should care. Is it possible to consolidate both scripts and eliminate this parameter? Can the uploader run auth logic from a local func and skip it if sessions file already in?

Reply:
The problem here is if the session file does not exist then Telethon python library will wait for user input (phone number, temporary code from Telegram, password – which cannot be passed as a parameters).
I'm not sure how current code that uploads audio is triggered, so if waiting for user input while running "uploader" is fine then I can take a look into that.

Reply:
hmm, this is odd. Are you saying this Telethon thingy is interactive? I'd expect the usual type of setup, i.e. setup account on tg side, get token(s), pass them into env and be done.

Reply:
Yes, it is interactive if session file does not exist. There is no way of getting "personal token", instead Telegram provides API ID and API Hash for building Telegram _client_. User is still required to type in their phone, sometimes 2FA password, and always temporary code from another Telegram client.

[By the way](https://core.telegram.org/api/obtaining_api_id):
> Due to excessive abuse of the Telegram API, all accounts that sign up or log in using unofficial Telegram API clients are automatically put under observation to avoid violations of the Terms of Service.

Reply:
I see. In this case, such a special preparation should be clearly documented in README. Looks like you already did for the direct run, but we need instruction for real-life, i.e. "docker exec -it ..." command. I also would suggest to combine scripts and add it as a command/subcommand to `uploader.py`

Reply:
> we need instruction for real-life, i.e. "docker exec -it ..." command

Added to `uploader/README.md`

> combine scripts and add it as a command/subcommand to uploader.py

Added flag `--auth_only` to `uploader.py`